src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.Parser.cs

```diff
@@ -0,0 +1,249 @@
+                node is not InvocationExpressionSyntax invocation
+                ? false
+                : IsBindCall(invocation) || IsConfigureCall(invocation) || IsGetCall(invocation);
+
+            private void ProcessBindCall(InvocationExpressionSyntax invocation, CancellationToken cancellationToken)
+            {
+                SemanticModel semanticModel = _compilation.GetSemanticModel(invocation.SyntaxTree);
+                IInvocationOperation operation = semanticModel.GetOperation(invocation, cancellationToken) as IInvocationOperation;
+
+                // We're looking for IConfiguration.Bind(object).
+                if (operation is IInvocationOperation { Arguments: { Length: 2 } arguments } &&
+                    operation.TargetMethod.IsExtensionMethod &&
+                    TypesAreEqual(_typeSymbolForIConfiguration, arguments[0].Parameter.Type) &&
+                    arguments[1].Parameter.Type.SpecialType == SpecialType.System_Object)
+                {
+                    IConversionOperation argument = arguments[1].Value as IConversionOperation;
+                    ITypeSymbol? type = ResolveType(argument)?.WithNullableAnnotation(NullableAnnotation.None);
+
+                    // TODO: do we need diagnostic for System.Object?
+                    if (type is not INamedTypeSymbol { } namedType ||
+                        namedType.SpecialType == SpecialType.System_Object ||
+                        namedType.SpecialType == SpecialType.System_Void)
+                    {
+                        return;
+                    }
+
+                    AddTargetConfigType(_typesForBindMethodGen, namedType, invocation.GetLocation());
+
+                    static ITypeSymbol? ResolveType(IOperation argument) =>
+                        argument switch
+                        {
+                            IConversionOperation c => ResolveType(c.Operand),
+                            IInstanceReferenceOperation i => i.Type,
+                            ILocalReferenceOperation l => l.Local.Type,
+                            IFieldReferenceOperation f => f.Field.Type,
+                            IMethodReferenceOperation m when m.Method.MethodKind == MethodKind.Constructor => m.Method.ContainingType,
+                            IMethodReferenceOperation m => m.Method.ReturnType,
+                            IAnonymousFunctionOperation f => f.Symbol.ReturnType,
+                            _ => null
+                        };
+                }
+            }
+
+            private void ProcessGetCall(InvocationExpressionSyntax invocation, CancellationToken cancellationToken)
+            {
+                SemanticModel semanticModel = _compilation.GetSemanticModel(invocation.SyntaxTree);
+                IInvocationOperation? operation = semanticModel.GetOperation(invocation, cancellationToken) as IInvocationOperation;
+
+                // We're looking for IConfiguration.Get<T>().
```

Comment:
>/ We're looking for IConfiguration.Get<T>() [](http://example.com/codeflow?start=17&length=43)

Same, will we need to handle the other Get methods? 

[https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.configurationbinder.get?view=dotnet-plat-ext-7.0#microsoft-extensions-configuration-configurationbinder-get(microsoft-extensions-configuration-iconfiguration-system-type](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.configurationbinder.get?view=dotnet-plat-ext-7.0#microsoft-extensions-configuration-configurationbinder-get(microsoft-extensions-configuration-iconfiguration-system-type))